filecoin-proofs/src/api/util.rs

```diff
@@ -83,8 +83,45 @@ pub(crate) fn get_p_aux<Tree: MerkleTreeTrait>(
     Ok(p_aux)
 }
 
-/// Instantiates t_aux from the specified cache_dir for access to  labels and tree_d, tree_c,
+/// Instantiates t_aux from default values for access to labels and tree_d, tree_c, tree_r_last
+/// store configs
+#[cfg(feature = "fixed-rows-to-discard")]
+// Silence Clippy warning in order to have the same return value as without this feature.
+#[allow(clippy::unnecessary_wraps)]
+pub(crate) fn get_t_aux<Tree: MerkleTreeTrait>(
+    cache_path: &Path,
+    sector_bytes: u64,
+) -> Result<TemporaryAux<Tree, DefaultPieceHasher>> {
+    trace!(
+        "Instantiating TemporaryAux from default values with cache_path at {:?}",
+        cache_path
+    );
+    let t_aux_path =
+        merkletree::store::StoreConfig::data_path(cache_path, &CacheKey::TAux.to_string());
+    if Path::new(&t_aux_path).exists() {
+        log::warn!(
```

Comment:
Everything looks good to me. I'm just wondering if this should fallback to `bincode::deserialize()` rather than a warning, or at least warn/fail if the taux on disk disagrees with the default taux values? But maybe that breaks something that I'm unaware of, and there's a good reason why this is just a warning.

Reply:
The idea is to fail early. If there is a `t_aux` file, then something with your setup/environment is wrong and you should really fix it. An implicit fallback could lead to hard to debug errors.

Reply:
If someone uses this code to do a sector update on a previously sealed sector, why fail if `t_aux` exists?

Reply:
We don't fail, we just warn that it is ignored. Of course things then might fail further down the line, hence the warning.

When we talked about it, I though we wanted a clear separating between "you run it with t_aux" and "you run it without t_aux". The "no t_aux" code path, never ever touches that file. In case there is a t_aux file, I would expect that it's more often accidental, rather then intentional.

But if we want to blur that distinction, I can make it read in the t_aux file. Though I'd prefer not doing that.

Reply:
> When we talked about it, I though we wanted a clear separating between "you run it with t_aux" and "you run it without t_aux". The "no t_aux" code path, never ever touches that file. In case there is a t_aux file, I would expect that it's more often accidental, rather then intentional.

Since the case of the 'no t_aux' doesn't touch the file, it shouldn't care that it's present.  Warning that it's present and unused is ok, but any failure caused by it being present will be a problem considering that just about every sealed and updated sector that exists today has it present.  There's no world where people can run this and be expected to re-seal everything, or manually remove every t_aux on disk in their operation to get the benefit.  Sure, someone may choose to start an operation with it enabled and that's the only legitimate case where 'no t_aux' files exist on disk.

Reply:
I was thinking that falling back to `bincode::deserialize()` would  make the `fixed-rows-to-discard` feature mean something like "for all sectors going forward seal using the default value for `rows_to_discard`, but for all previously sealed sectors (e.g. during sector updates) use that sector's t_aux file (if it exists)".

Reply:
Alright, I've added the fallback. I squashed the commits as I wanted to change the commit to reflect the reality. The diff between before and after the force push is pretty clean though: https://github.com/filecoin-project/rust-fil-proofs/compare/2ee6eda8e577728a3f8dcf3e732859e1442ec201..b02359f97effeca0f84ff95543284527b04975d4

Reply:
Thanks for making the change. Everything looks good to me